catalog/tests/test_views.py

```diff
@@ -0,0 +1,258 @@
+from django.test import TestCase
+
+# Create your tests here.
+from django.test import TestCase
+from django.urls import reverse
+
+from catalog.models import Book
+
+class BookListViewTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        # Create 5 books for pagination tests
+        number_of_books = 5
+
+        for book_id in range(number_of_books):
+            Book.objects.create(
+                title=f'Art of Making Money {book_id}',
+                author=f'Ronald',
+            )
+
+    def test_view_url_exists_at_desired_location(self):
+        response = self.client.get('/catalog/books/')
+        self.assertEqual(response.status_code, 200)
+
+    def test_view_url_accessible_by_name(self):
+        response = self.client.get(reverse('books'))
+        self.assertEqual(response.status_code, 200)
+
+    def test_view_uses_correct_template(self):
+        response = self.client.get(reverse('books'))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'catalog/book_list.html')
+
+    def test_pagination_is_three(self):
+        response = self.client.get(reverse('books'))
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('is_paginated' in response.context)
+        self.assertTrue(response.context['is_paginated'] == True)
+        self.assertEqual(len(response.context['book_list']), 3)
+
+    def test_lists_all_books(self):
+        # Get second page and confirm it has (exactly) remaining 3 items
+        response = self.client.get(reverse('books')+'?page=2')
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('is_paginated' in response.context)
+        self.assertTrue(response.context['is_paginated'] == True)
+        self.assertEqual(len(response.context['book_list']), 2)
+
+
+import datetime
+
+from django.utils import timezone
+from django.contrib.auth.models import User # Required to assign User as a borrower
+
+from catalog.models import BookCopy, Book
+
```

Comment:
It's better to have import at top (one place). here you are also reimporting book that should not be done.